contracts/core/EntryPoint.sol

```diff
@@ -39,6 +39,15 @@ contract EntryPoint is IEntryPoint, StakeManager, NonceManager, ReentrancyGuard
      */
     uint256 public constant SIG_VALIDATION_FAILED = 1;
 
+    /// @inheritdoc OpenZeppelin.IERC165
+    function supportsInterface(bytes4 interfaceId) public view virtual override(OpenZeppelin.ERC165, OpenZeppelin.IERC165) returns (bool) {
+        return interfaceId == (type(IEntryPoint).interfaceId ^ type(IStakeManager).interfaceId ^ type(INonceManager).interfaceId) ||
```

Comment:
1. add a comment here that solidity ignores the "inheritance" of interfaces.
2. need to add the IERC165 Id itself (since it IS part of the IEntryPoint as we define it